### PR TITLE
LCPD for general ranges of bias, amplitudes, tip charge and cantilever stiffness

### LINKED ISSUE #245: Make KPAFM work with general bias voltage

### LINKED ISSUE #272: Make KPFM work for a range of amplitudes, probe-particle stiffness parameters etc.
Currently, LCPD is always calculated only for one amplitude, one probe-particle stifness, and one tip charge. If ranges of parameters are specified instead of single values, df will be calculated for the whole specified range(s) but LCPD only for the last combination of values, without warning the user that this is the case. I want to make KPFM work with the ranges of parameters (`--arange`, `--krange`, `--crange`) in the same way AFM does.

### PR DESCRIPTION
**Pull request by mondracek**
Resolves #272 and resolves #245 too.

Note that from now on, the files `LCPD.xsf` and `*LCPD*.png` will be placed in a `Q?.??K?.??/Amp?.??/` rather than directly in the working directory, because, in general, more LCPD results for different _Q_, _k_, _A_ can be calculated in one run (consequence of implementing issue #272).
No restriction on `--Vrange` now (implementing issue #245), except that of course at least 3 different bias values are needed to evaluate LCPD . Regression of quadratic polynomial (for the Kelvin parabola) with the least-square-of-error criterion will be used when more than three values of bias are available. The bias range in `ppafm-plot-results --Vrange` still needs to be compatible with the range in `ppafm-relaxed-scan --Vrange`, of course.

### CODE REVIEW

**Comment by mondracek** [comment]
Tested on CH3Br, and FFPB. The resulting `*LCPD*.png` files are virtually the same as with the previous implementation. They are not exactly bit-wise identical if different `--Vrange` are used, but that is expected.

**Comment by mondracek** [comment]
Continuous integration files with test-package v 3.9 and higher. Probably nothing to do with the part of the code I have modified. @yakutovicha, you probably understand the best how this continuous integration is supposed to work and how to manage it. Can you give me a hint what may be going on?

**Comment by mondracek** [comment]
What I have found about the continuous-integration tests:
* They fail at the second version of python that is going to be tested, no matter which version it is
* Therefore, if I modify `ci.yml` to test only one version of python, the test passes.
* The failing test fails at the `Run pytest` stage, the error message (at lines 31-33) is the following
   - 31 ` /home/runner/work/_temp/6ab19913-178e-4b8f-92bf-dae56f3545bf.sh: line 1:  1858 Segmentation fault      (core dumped) PPAFM_RECOMPILE=1 pytest tests examples -v --cov --cov-report json`
   - 32 `examples/PTCDA_single/example_ptcda.py::example_ptcda_single`
   - 33 `Error: Process completed with exit code 139.`
 * Nothing to do with my code changes. The multiple-version tests failed even when I tried to upload a `test` branch which was an exact copy of the current `main` branch.

What happened as a side effect of me finding out and what I am going to do now:
 * I have created two new commits on GitHub by editing `ci.yml`, solely for the debugging purpose. I am going to force-push this branch again from my local repository *now* in order to remove this commits.
 * I am going to create a new issue to report this problem, as it is in fact not specific to this particular PR.

**Comment by NikoOinonen** [comment]
@mondracek I merged the PR for fixing the CI tests. If you merge main into this branch, the tests should pass.

**Comment by mondracek** [comment]
Sorry, I rebased and force-pushed instead of merging with main, which could be undesired in some cases and I am going to avoid it next time, but that's a side point here ... The final result should be the same. What is unfortunately the important point is that
_The tests are still failing, even with only examples/PTCDA_Hartree_dz2/ being tested._

**Comment by NikoOinonen** [comment]
@mondracek There is something odd going on with the tests. I simply reran them and they passed, so it seems not to be deterministic. If you are confident that this PR cannot be influencing this, then I would just go ahead with it now.

**Comment by mondracek** [comment]
Okay. I will wait for one approving review before merging. Preferably from @aureliojgc , as he knows the details of the LCPD calculations, but if Aurelio does not have time for it, anyone who feels what I did makes sense can approve.

**Review by aureliojgc — APPROVED** [review]
Everything seems fine for the merging. No problem from my side and will test it more thoroughly against previous projects in the following days

**Comment by NikoOinonen** [comment]
@mondracek Shouldn't this be merged or is there some problem?

**Comment by mondracek** [comment]
I thought I should wait for @aureliojgc to finish the tests he mentions above but then I decided to just go ahead and merge it.

**Pull request merged**
